### ISSUE TITLE
24 hour time format support

### ISSUE THREAD

**Issue by Christilut** [zhukov/webogram#579]
A setting (or even better, if it detects the correct settings based on locale or a browser setting) that puts time stamps in the 24 hour format would be great.

**Comment by zhukov** [comment]
We have 24hr support. It is enabled by selected languages. English is considered as US, so the date is in am/pm format by default.

**Comment by Christilut** [comment]
Okay, I've put mine on Dutch for the preferred 24hr setting. 
I guess it's not a real issue anymore but some users (like myself) prefer everything in English but with 24hr as timestamps.

But since there is not much localized text going, I wouldn't really bother creating an option for this.

**Comment by mcepl** [comment]
> We have 24hr support. It is enabled by selected languages. English is considered as US, so the date is in am/pm format by default.

So, could we get some other English (e.g., `en_UK`) which has 24h format?

**Comment by Lohoris** [comment]
I agree: binding a language to a hour format does not make much sense, couldn't we just have two separate settings?

**Comment by arkhnchul** [comment]
agree with it. "English considered as US" is totally wrong in the locale's case.

**Comment by ahstro** [comment]
This goes along with date formats as well. Currently, they're `mm/dd/yyyy` for me, but I'd very much prefer `dd/mm/yyyy`. Not sure if these are set based on language as well?

**Comment by Tanoh** [comment]
A simple workaround that I use is a bookmarklet that I run before the page has finished loading:

```javascript
javascript:(function()%7Bd%3Dangular.element(document.body).injector().get('%24locale').DATETIME_FORMATS%3Bd.medium%3D"d` MMM y H%3Amm%3Ass"%3Bd.mediumDate%3D"d MMM y"%3Bd.mediumTime%3D"H%3Amm%3Ass"%3Bd.short%3D"yyyy-MM-dd H%3Amm"%3Bd.shortDate%3D"yyyy-MM-dd"%3Bd.shortTime%3D"H%3Amm"%7D)()
```

But it's a bit annoying to have to remember to click it, and it would be a lot nicer to have proper support for time and date formats.

**Comment by ahstro** [comment]
> it's a bit annoying to have to remember to click it

@Tanoh Couldn't you run it with [dotjs](https://github.com/rlr/dotjs-addon)?

**Comment by Ryuno-Ki** [comment]
I assume, the easiest way out would defer the formatting to [angular-moment](https://www.npmjs.com/package/angular-moment).

Date formats are defined [in angular i18n](https://github.com/zhukov/webogram/tree/82aec7a6daea93631b9eb14a89248d2b9580a906/app/vendor/angular/i18n).
The formatting calls happen in several controllers (search for `$filter("date")`) instead of templates.

**Comment by stek29** [comment]
@Tanoh Can't you make userscript of that?

**Comment by emilfihlman** [comment]
Please fix this, it is annoying as heck

**Comment by Saucy** [comment]
Any update on this feature?

**Comment by emilfihlman** [comment]
Yeah should be super simple and easy to add, please do so.

**Comment by Ryuno-Ki** [comment]
PRs welcome? ;-)
